packages/ssr-web/app/components/common/card-space-user-page/index.ts

```diff
@@ -0,0 +1,94 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import config from '@cardstack/ssr-web/config/environment';
+import { generateMerchantPaymentUrl } from '@cardstack/cardpay-sdk';
+import { action } from '@ember/object';
+import { PaymentLinkMode } from '../payment-link';
+import { inject as service } from '@ember/service';
+import UA from '@cardstack/ssr-web/services/ua';
+import Subgraph from '@cardstack/ssr-web/services/subgraph';
+import * as Sentry from '@sentry/browser';
+
+interface CardSpaceUserPageArgs {
+  model: {
+    did: string;
+    id: string;
+    name: string;
+    backgroundColor: string;
+    textColor: string;
+  };
+}
+
+export default class CardSpaceUserPage extends Component<CardSpaceUserPageArgs> {
+  @service('ua') declare UAService: UA;
+  @tracked paymentLinkMode: PaymentLinkMode = 'link';
+  @tracked address: string | null = null;
+  @tracked addressFetchingError: string | null = null;
+  @service declare subgraph: Subgraph;
+  defaultAddressFetchingErrorMsg =
+    'We ran into an issue while generating the payment request link. Please reload the page and try again. If the issue persists, please contact support.';
+
+  @action setInitialPaymentLinkMode() {
+    if (!this.canDeepLink) {
+      this.paymentLinkMode = 'qr-non-mobile';
+    }
+  }
+
+  @action setPaymentLinkMode(mode: PaymentLinkMode) {
+    this.paymentLinkMode = mode;
+  }
+
+  get canDeepLink() {
+    return this.UAService.isIOS() || this.UAService.isAndroid();
+  }
+
+  @action async loadAddress() {
+    this.addressFetchingError = null;
+
+    let address;
+
+    try {
+      let queryResult = await this.subgraph.query(
+        config.chains.layer2,
+        `query($did: String!) {
+            merchantSafes(where: { infoDid: $did }) {
+              id
+            }
+          }`,
+        {
+          did: this.args.model.did,
+        }
+      );
+
+      address = queryResult?.data?.merchantSafes[0]?.id;
+    } catch (e) {
+      this.addressFetchingError = this.defaultAddressFetchingErrorMsg;
+      Sentry.captureException(e);
+    }
+
+    if (address) {
+      this.address = address;
+    } else {
+      this.addressFetchingError = this.defaultAddressFetchingErrorMsg;
```

Comment:
This message is likely not very appropriate for this case. To tune this, we need to know what are the possible values the subgraph can return when there is no merchant safe id included.

1. Subgraph returns empty merchants 
- Odd edge case where safe is deleted or can't be associated for some reason? Should we handle this case?

2. Subgraph returns something like `indexing_error` (as we observed recently)
- We can probably keep that error message

3. Anything else?

Reply:
I can’t think of anything else at least, but this is easily revisited if we find other failures.

Reply:
I think this is ok, is it useful to have this logged to sentry?

Reply:
I think it would be useful. Will add that!

Reply:
Added!